server/src/main/java/org/elasticsearch/action/search/ExpandSearchPhase.java

```diff
@@ -65,6 +65,9 @@ private boolean isCollapseRequest() {
 
     @Override
     public void run() {
+        //TODO we can't really tell how many shards this runs on, we could say 0 whenever nothing has to happen, but what otherwise?
+        //TODO shall we always mark the phase started even when it does nothing?
+        context.getTask().getStatus().phaseStarted(getName(), -1);
```

Comment:
good point. Expand is a bit of a different phase compared to all the others we have. It may be invoked despite it does nothing (I think all the others phases are only invoked when they need to do something). Also, it sends a multi_search request. I am fine with counting number of operations as `1`, though also TaskInfo does not fit in a SearchResponse hence we would end up with a phase that gets started and completed but that expected operation does not really report back by calling either shardProcessed or shardFailed. That is why I would still leave those two methods referring to shards, because whenever phases don't work with shards, there is no SearchPhaseResult and those methods cannot be invoked.

I also have a problem because I currently end the phase in executeNextPhase, which is problematic when a phase is invoked yet it does nothing, hence it has not started but executeNextPhase will always complete it. I removed the two old TODOs and added a new one around this. We may be able to refactor executeNextPhase on master to improve this.